`indent`: `outerIIFEBody` option not using a multiplier, warning on non-IIFEs?

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**What version of ESLint are you using?**
v3.0.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```sh
eslint --rule indent:[2,2,{outerIIFEBody:1}] file.js
```

**What did you do? Please include the actual source code causing the issue.**
https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/test/test-base.js
```js
import fs from 'fs';
import path from 'path';
import test from 'tape';

import index from '../';

const files = { index };

fs.readdirSync(path.join(__dirname, '../rules')).forEach(name => {
  files[name] = require(`../rules/${name}`); // eslint-disable-line global-require
});

Object.keys(files).forEach(name => {
  const config = files[name];

  test(`${name}: does not reference react`, t => {
    t.plan(2);

    // scan plugins for react and fail if it is found
    const hasReactPlugin = Object.prototype.hasOwnProperty.call(config, 'plugins') &&
      config.plugins.indexOf('react') !== -1;
    t.notOk(hasReactPlugin, 'there is no react plugin');

    // scan rules for react/ and fail if any exist
    const reactRuleIds = Object.keys(config.rules)
      .filter(ruleId => ruleId.indexOf('react/') === 0);
    t.deepEquals(reactRuleIds, [], 'there are no react/ rules');
  });
});
```

**What did you expect to happen?**
No warnings.

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
$PWD/packages/eslint-config-airbnb-base/test/test-base.js
  10:3  error  Expected indentation of 1 space character but found 2  indent
  14:3  error  Expected indentation of 1 space character but found 2  indent
  16:3  error  Expected indentation of 1 space character but found 2  indent
```

There are two problems here:
1) the warned lines are not actually file-level IIFEs - the file doesn't have an IIFE at all.
2) `outerIIFEBody` is set to 1 - this should be a multiplier of 1 on the indentation config of "2 spaces" - the errors indicate that it wants "1 space" which should **never** be required when the base indent is "2".

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -369,18 +369,23 @@ module.exports = {
 
 		/**
          * Check to see if the node is a file level IIFE
-         * @param {ASTNode} node node to check
+         * @param {ASTNode} node The function node to check.
          * @returns {boolean} True if the node is the outer IIFE
          */
         function isOuterIIFE(node) {
-            return node && node.type === "CallExpression" &&
-                node.parent && node.parent.type === "ExpressionStatement" &&
-                node.parent.parent && node.parent.parent.type === "Program";
+            var parent = node.parent;
+
+            return (
+                parent.type === "CallExpression" &&
+                parent.callee === node &&
+                parent.parent.type === "ExpressionStatement" &&
+                parent.parent.parent && parent.parent.parent.type === "Program"
+            );
         }
 
         /**
          * Check indent for function block content
-         * @param {ASTNode} node node to examine
+         * @param {ASTNode} node A BlockStatement node that is inside of a function.
          * @returns {void}
          */
         function checkIndentInFunctionBlock(node) {
@@ -433,8 +438,8 @@ module.exports = {
             // is the outer IIFE and that option is enabled.
             var functionOffset = indentSize;
 
-            if (options.outerIIFEBody !== null && isOuterIIFE(calleeNode.parent)) {
-                functionOffset = options.outerIIFEBody;
+            if (options.outerIIFEBody !== null && isOuterIIFE(calleeNode)) {
+                functionOffset = options.outerIIFEBody * indentSize;
             }
             indent += functionOffset;
 

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -1139,6 +1139,14 @@ ruleTester.run("indent", rule, {
         },
         {
             code:
+            "fs.readdirSync(path.join(__dirname, '../rules')).forEach(name => {\n" +
+            "  files[name] = foo;\n" +
+            "});",
+            options: [2, { outerIIFEBody: 0 }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code:
             "(function(){\n" +
             "function foo(x) {\n" +
             "  return x + 1;\n" +
@@ -1149,9 +1157,9 @@ ruleTester.run("indent", rule, {
         {
             code:
             "(function(){\n" +
-            "  function foo(x) {\n" +
-            "      return x + 1;\n" +
-            "  }\n" +
+            "        function foo(x) {\n" +
+            "            return x + 1;\n" +
+            "        }\n" +
             "})();",
             options: [4, { outerIIFEBody: 2 }]
         },
@@ -2137,7 +2145,7 @@ ruleTester.run("indent", rule, {
             "    }\n" +
             "})();",
             options: [4, { outerIIFEBody: 2 }],
-            errors: expectedErrors([[2, 2, 4, "FunctionDeclaration"]])
+            errors: expectedErrors([[2, 8, 4, "FunctionDeclaration"]])
         },
         {
             code:

